Assets/Scripts/Camera/SpawnTargets.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Camera
+{
+    public class SpawnTargets : MonoBehaviour
+    {
+        public List<Transform> targetPrefabs;
+        public Transform parentObject;
+
+        private UnityEngine.Camera _camera;
+        private void Awake()
+        {
+            _camera = GetComponent<UnityEngine.Camera>();
+        }
+
+        private void Update()
+        {
+            if (!Input.GetMouseButtonDown(0))
+            {
+                return;
+            }
+        
+            var ray = _camera.ScreenPointToRay(Input.mousePosition);
+            Physics.Raycast(ray, out var hit);
```

Comment:
you should always check for the result of Raycast (true/false). If you shoot at skybox, you do not hit anything. But still, a character gets spawned in the default position (zero). This seems like a bug rather than expected behavior